testing/dart/fragment_shader_test.dart

```diff
@@ -64,6 +67,38 @@ void main() {
     _expectShaderRendersGreen(shader);
   });
 
+  test('simple iplr shader can be re-initialized', () async {
+    final FragmentProgram program = FragmentProgram.fromAsset(
+      'functions.frag.iplr',
+    );
+    final Shader shader = program.shader(
+      floatUniforms: Float32List.fromList(<double>[1]),
+    );
+
+    final developer.ServiceProtocolInfo info = await developer.Service.getInfo();
+
+    if (info.serverUri == null) {
+      fail('This test must not be run with --disable-observatory.');
```

Comment:
Non-debug should have asserts disabled, so the simplest thing is probably just to return from the test immediately if asserts are disabled.

Reply:
Turns out we have a special test directory for this